### PR TITLE
Improved Exception Dialog with Copy Paste

### LINKED ISSUE #1861: Improved exception dialog with copy & paste or direct error reporting
#### Description

When Vorta crashed due to a python exception, one gets a dialogue box telling that one shall report the issue.

Problems:
- copy and paste does not work from there, so one has to manually type the whole stuff one is seeing in that box.
- the dialogue box contents are rather ugly / badly formatted

#### Environment

- OS: macOS Ventura
- Vorta version: 0.9.1beta3
- Installed from: .dmg binary download
- Borg version: 1.2.6

### PR DESCRIPTION
**Pull request by shivansh02**
### Description

Exception dialog improvements:
-better UI
-copy paste functionality
-detailed error with system information
-report to github functionality

### Related Issue

Fixes #1861

### Motivation and Context

Exception Dialog currently has bad UI and no copy paste functionality.
### Screenshots (if appropriate):
Screenshot from 2024-03-31 08-01-44

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

File: src/vorta/views/exception_dialog.py
Comment (diivi) [inline src/vorta/views/exception_dialog.py:44]: Consider giving these labels appropriate names.

File: src/vorta/views/exception_dialog.py
Comment (diivi) [inline src/vorta/views/exception_dialog.py:46]: maybe add a copy icon here for quick visual assistance.

File: src/vorta/views/exception_dialog.py
Comment (diivi) [inline src/vorta/views/exception_dialog.py:31]: redundant intermediate variable.

**Review by diivi — COMMENTED** [review]
Looks like a nice improvement over the error dialog that we have currently, thanks!

File: src/vorta/views/exception_dialog.py
Comment (diivi) [inline src/vorta/views/exception_dialog.py:23]: Not sure if this is necessary.
Looks good for me too otherwise:

image

**Comment by diivi** [comment]
I suppose `.github/ISSUE_TEMPLATE/bug_report.md` and `.github/ISSUE_TEMPLATE/bug_form.yaml` will also need to be changed as a part of this. Will let the maintainers decide.

**Comment by shivansh02** [comment]
@m3nu @ThomasWaldmann Can you please take a look?

File: src/vorta/assets/icons/alert.svg
Comment (real-yfprojects) [inline src/vorta/assets/icons/alert.svg:1]: Where did you source this image from?

**Comment by real-yfprojects** [comment]
> I suppose `.github/ISSUE_TEMPLATE/bug_report.md` and `.github/ISSUE_TEMPLATE/bug_form.yaml` will also need to be changed as a part of this.

This can be done once this is merged.

File: src/vorta/views/exception_dialog.py
Comment (real-yfprojects) [inline src/vorta/views/exception_dialog.py:57]: What is that call needed for?

**Review by real-yfprojects — COMMENTED** [review]
Nice addition!

File: src/vorta/views/exception_dialog.py
Comment (ThomasWaldmann) [reply src/vorta/views/exception_dialog.py:23]: Maybe use a format string for the datetime, so it doesn't put the subsecond decimals there.

About the "Log:":
- that's a traceback and guess you did not get it from the log, but directly within python.
- but, vorta (and also borg) do write a log, which might have some additional interesting infos, which you do not yet include. maybe a link that opens the correct log file so that the user can easily c&p stuff from there would be useful here.

File: src/vorta/assets/icons/alert.svg
Comment (shivansh02) [reply src/vorta/assets/icons/alert.svg:1]: Wikipedia commons: [https://en.m.wikipedia.org/wiki/File:Warning.svg](url)

File: src/vorta/views/exception_dialog.py
Comment (shivansh02) [reply src/vorta/views/exception_dialog.py:57]: It clears the clipboard before setting new text, but it seems to be working fine without this line also. Will remove it :)

File: src/vorta/assets/icons/alert.svg
Comment (real-yfprojects) [reply src/vorta/assets/icons/alert.svg:1]: If I understand correctly the image file is released into the public domain and we are free to relicence it?
We should also consider using the [corresponding fork awesome icon](https://forkaweso.me/Fork-Awesome/icon/exclamation-triangle/) for consistency with the other icons in the GUI.

**Comment by m3nu** [comment]
Needs rebase and an icon from the sources we usually use https://vorta.borgbase.com/contributing/#icons

**Comment by shivansh02** [comment]
> Needs rebase and an icon from the sources we usually use https://vorta.borgbase.com/contributing/#icons

@m3nu Done, and the checks are passing as well. Can we merge this?

**Review by m3nu — APPROVED** [review]

**Pull request merged**
